# Classify HFpEF

In [1]:
import pandas as pd
import numpy as np

from preproc import get_hfpef_200, convert2np

In [2]:
lbl_colname= ['Death', 'CV death', 'Major cardiac events']

## Clean Data

In [3]:
data_df = get_hfpef_200()

In [4]:
data_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 49 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Age                   200 non-null    int64  
 1   Sex                   200 non-null    int64  
 2   Cr                    200 non-null    float64
 3   GFR                   200 non-null    float64
 4   CKD stage             200 non-null    int64  
 5   smoke                 200 non-null    float64
 6   BMI                   200 non-null    float64
 7   BSA                   200 non-null    float64
 8   DM                    200 non-null    int64  
 9   Insulin               200 non-null    int64  
 10  AF                    200 non-null    int64  
 11  SBP                   200 non-null    int64  
 12  DBP                   200 non-null    int64  
 13  MAP                   200 non-null    float64
 14  PP                    200 non-null    int64  
 15  1/2SBP                2

In [5]:
data_df.describe()

,Age,Sex,Cr,GFR,CKD stage,smoke,BMI,BSA,DM,Insulin,...,E/A,Mitral E/e',TR Vmax,RWT,LV mass index,LVH,LAVI,LACI,LA diameter,LVEF
count,200.000000,200.000000,200.000000,200.000000,200.000000,200.000000,200.000000,200.000000,200.000000,200.000000,...,200.000000,200.000000,200.000000,200.000000,200.000000,200.000000,200.000000,200.000000,200.000000,200.000000
mean,72.045000,1.680000,1.711100,52.293250,2.810000,0.105000,26.738830,1.632980,0.560000,0.215000,...,0.741571,17.300650,1.890100,0.648305,115.640950,2.640000,17.428481,3.211636,16.797793,61.434500
std,11.404529,0.467647,1.534721,25.188572,1.043718,0.307323,6.636857,0.228277,0.497633,0.411853,...,0.606755,8.602117,1.443655,0.388423,52.778994,1.896943,52.106319,3.988946,18.158083,8.420635
min,31.000000,1.000000,0.550000,2.000000,1.000000,0.000000,15.400000,1.120000,0.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,-100.000000,0.000000,0.000000,0.000000
25%,65.000000,1.000000,0.887500,34.750000,2.000000,0.000000,22.200000,1.485000,0.000000,0.000000,...,0.500000,11.600000,0.000000,0.400000,86.725000,1.000000,0.000000,0.000000,3.700000,56.000000
50%,74.000000,2.000000,1.140000,54.000000,3.000000,0.000000,26.300000,1.600000,1.000000,0.000000,...,0.700000,16.300000,2.460000,0.555000,113.150000,2.500000,32.600000,2.479869,4.500000,61.000000
75%,80.000000,2.000000,1.792500,70.250000,3.000000,0.000000,29.925000,1.800000,1.000000,0.000000,...,1.000000,22.100000,3.002500,0.865000,141.305000,4.000000,48.425000,5.111374,37.692250,66.000000
max,97.000000,2.000000,10.000000,108.000000,5.000000,1.000000,55.900000,2.300000,1.000000,1.000000,...,4.100000,42.200000,4.550000,2.300000,356.600000,7.000000,140.800000,23.665480,59.919000,82.900000


## Get Cleaned Data

In [6]:
data_df_nomiss = get_hfpef_200(nomiss=True)

In [8]:
data_df_nomiss.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 80 entries, 1 to 199
Data columns (total 49 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Age                   80 non-null     int64  
 1   Sex                   80 non-null     int64  
 2   Cr                    80 non-null     float64
 3   GFR                   80 non-null     float64
 4   CKD stage             80 non-null     int64  
 5   smoke                 80 non-null     float64
 6   BMI                   80 non-null     float64
 7   BSA                   80 non-null     float64
 8   DM                    80 non-null     int64  
 9   Insulin               80 non-null     int64  
 10  AF                    80 non-null     int64  
 11  SBP                   80 non-null     int64  
 12  DBP                   80 non-null     int64  
 13  MAP                   80 non-null     float64
 14  PP                    80 non-null     int64  
 15  1/2SBP                80

In [9]:
data_df_nomiss.describe()

,Age,Sex,Cr,GFR,CKD stage,smoke,BMI,BSA,DM,Insulin,...,E/A,Mitral E/e',TR Vmax,RWT,LV mass index,LVH,LAVI,LACI,LA diameter,LVEF
count,80.000000,80.000000,80.000000,80.000000,80.000000,80.000000,80.000000,80.000000,80.00000,80.000000,...,80.000000,80.000000,80.000000,80.000000,80.0000,80.000000,80.000000,80.000000,80.000000,80.000000
mean,72.212500,1.700000,1.662125,54.480500,2.737500,0.062500,26.156735,1.609162,0.58750,0.250000,...,0.961500,19.038625,1.902625,0.672113,121.0900,2.925000,24.565556,5.152243,18.100662,62.260000
std,11.623518,0.461149,1.486146,27.486385,1.110864,0.243589,6.292129,0.198474,0.49539,0.435745,...,0.476238,8.867438,1.422817,0.376618,45.8851,1.826493,45.879166,4.377082,17.768611,7.273406
min,31.000000,1.000000,0.640000,2.000000,1.000000,0.000000,15.400000,1.120000,0.00000,0.000000,...,0.400000,4.290000,0.000000,0.230000,53.8000,0.000000,-95.629630,0.000000,2.100000,50.000000
25%,65.000000,1.000000,0.855000,30.250000,2.000000,0.000000,21.825000,1.465000,0.00000,0.000000,...,0.600000,12.475000,0.000000,0.420000,90.2000,1.000000,23.425000,2.556917,3.975000,57.825000
50%,74.500000,2.000000,1.115000,57.500000,3.000000,0.000000,25.450000,1.600000,1.00000,0.000000,...,0.900000,17.500000,2.405000,0.550000,113.8500,3.000000,33.850000,4.403986,4.500000,61.250000
75%,79.000000,2.000000,1.900000,77.250000,3.250000,0.000000,29.825000,1.705000,1.00000,0.250000,...,1.125000,23.350000,3.050000,0.860000,135.4000,4.000000,47.750000,6.716609,37.069000,66.250000
max,92.000000,2.000000,10.000000,108.000000,5.000000,1.000000,44.400000,2.000000,1.00000,1.000000,...,3.200000,42.200000,4.550000,2.300000,356.6000,7.000000,107.700000,23.665480,59.919000,80.000000


## Random Forest

In [10]:
from sklearn.model_selection import GridSearchCV
from sklearn.metrics import classification_report

from sklearn.ensemble import RandomForestClassifier
from sklearn.neural_network import MLPClassifier

from sklearn.model_selection import cross_val_score, cross_val_predict, cross_validate 
from sklearn.metrics import classification_report

In [11]:
def hfpef_rf(X, y, feature_list):
    # Set the parameters by cross-validation
    tuned_parameters = [{'max_depth': [5,10,20],
                         'n_estimators': [100, 200, 400]}]

    rf = RandomForestClassifier(random_state=0)

    for i in range(y.shape[1]):
        print('==================================================')
        print('Prediction of ' + lbl_colname[i])

        y_true = y[:,i]

        clf = GridSearchCV(rf, tuned_parameters, scoring='f1_macro', cv=5)
        clf.fit(X, y_true)

        print("Best parameters set found on development set:")
        print()
        print(clf.best_params_)
        print()
        print("Grid scores on development set:")
        print()
        means = clf.cv_results_['mean_test_score']
        stds = clf.cv_results_['std_test_score']
        for mean, std, params in zip(means, stds, clf.cv_results_['params']):
            print("%0.3f (+/-%0.03f) for %r"
                  % (mean, std * 2, params))
        print()

        y_pred = cross_val_predict(clf.best_estimator_, X, y_true, cv=5)

        print(f"count {sum(y_pred)}/{len(y_pred)}")
        print("Detailed classification report:")
        print()
        print(classification_report(y_true, y_pred))
        print()

        for i,v in sorted(zip(feature_list, clf.best_estimator_.feature_importances_), key=lambda tup: tup[1], reverse=True)[:20]:
            print('Feature: %s, Score: %.5f' % (i,v))

### All Features

In [12]:
X, y, feature_list = convert2np(data_df, lbl_colname)

In [13]:
hfpef_rf(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'max_depth': 5, 'n_estimators': 100}

Grid scores on development set:

0.458 (+/-0.006) for {'max_depth': 5, 'n_estimators': 100}
0.458 (+/-0.006) for {'max_depth': 5, 'n_estimators': 200}
0.458 (+/-0.006) for {'max_depth': 5, 'n_estimators': 400}
0.456 (+/-0.012) for {'max_depth': 10, 'n_estimators': 100}
0.458 (+/-0.006) for {'max_depth': 10, 'n_estimators': 200}
0.458 (+/-0.006) for {'max_depth': 10, 'n_estimators': 400}
0.456 (+/-0.012) for {'max_depth': 20, 'n_estimators': 100}
0.458 (+/-0.006) for {'max_depth': 20, 'n_estimators': 200}
0.458 (+/-0.006) for {'max_depth': 20, 'n_estimators': 400}

count 0/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.84      1.00      0.92       169
           1       0.00      0.00      0.00        31

    accuracy                           0.84       200
   macro avg       0.42      0.50      0.46       200
weight

C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Best parameters set found on development set:

{'max_depth': 5, 'n_estimators': 100}

Grid scores on development set:

0.486 (+/-0.005) for {'max_depth': 5, 'n_estimators': 100}
0.486 (+/-0.005) for {'max_depth': 5, 'n_estimators': 200}
0.486 (+/-0.005) for {'max_depth': 5, 'n_estimators': 400}
0.486 (+/-0.005) for {'max_depth': 10, 'n_estimators': 100}
0.486 (+/-0.005) for {'max_depth': 10, 'n_estimators': 200}
0.486 (+/-0.005) for {'max_depth': 10, 'n_estimators': 400}
0.486 (+/-0.005) for {'max_depth': 20, 'n_estimators': 100}
0.486 (+/-0.005) for {'max_depth': 20, 'n_estimators': 200}
0.486 (+/-0.005) for {'max_depth': 20, 'n_estimators': 400}

count 0/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.94      1.00      0.97       189
           1       0.00      0.00      0.00        11

    accuracy                           0.94       200
   macro avg       0.47      0.50      0.49       200
weighted avg       0.89   

C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Best parameters set found on development set:

{'max_depth': 5, 'n_estimators': 400}

Grid scores on development set:

0.663 (+/-0.056) for {'max_depth': 5, 'n_estimators': 100}
0.665 (+/-0.097) for {'max_depth': 5, 'n_estimators': 200}
0.682 (+/-0.100) for {'max_depth': 5, 'n_estimators': 400}
0.636 (+/-0.043) for {'max_depth': 10, 'n_estimators': 100}
0.655 (+/-0.058) for {'max_depth': 10, 'n_estimators': 200}
0.639 (+/-0.069) for {'max_depth': 10, 'n_estimators': 400}
0.624 (+/-0.062) for {'max_depth': 20, 'n_estimators': 100}
0.645 (+/-0.067) for {'max_depth': 20, 'n_estimators': 200}
0.641 (+/-0.064) for {'max_depth': 20, 'n_estimators': 400}

count 110/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.70      0.64      0.67        99
           1       0.67      0.73      0.70       101

    accuracy                           0.69       200
   macro avg       0.69      0.68      0.68       200
weighted avg       0.69 

### Selected Features

In [14]:
X, y, feature_list = convert2np(data_df, lbl_colname, selected_feat=True)

In [15]:
hfpef_rf(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'max_depth': 5, 'n_estimators': 200}

Grid scores on development set:

0.455 (+/-0.018) for {'max_depth': 5, 'n_estimators': 100}
0.456 (+/-0.012) for {'max_depth': 5, 'n_estimators': 200}
0.455 (+/-0.018) for {'max_depth': 5, 'n_estimators': 400}
0.455 (+/-0.018) for {'max_depth': 10, 'n_estimators': 100}
0.455 (+/-0.018) for {'max_depth': 10, 'n_estimators': 200}
0.455 (+/-0.018) for {'max_depth': 10, 'n_estimators': 400}
0.455 (+/-0.018) for {'max_depth': 20, 'n_estimators': 100}
0.455 (+/-0.018) for {'max_depth': 20, 'n_estimators': 200}
0.453 (+/-0.018) for {'max_depth': 20, 'n_estimators': 400}

count 1/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.84      0.99      0.91       169
           1       0.00      0.00      0.00        31

    accuracy                           0.84       200
   macro avg       0.42      0.50      0.46       200
weight

C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Best parameters set found on development set:

{'max_depth': 5, 'n_estimators': 200}

Grid scores on development set:

0.673 (+/-0.125) for {'max_depth': 5, 'n_estimators': 100}
0.688 (+/-0.097) for {'max_depth': 5, 'n_estimators': 200}
0.677 (+/-0.086) for {'max_depth': 5, 'n_estimators': 400}
0.687 (+/-0.079) for {'max_depth': 10, 'n_estimators': 100}
0.666 (+/-0.135) for {'max_depth': 10, 'n_estimators': 200}
0.665 (+/-0.133) for {'max_depth': 10, 'n_estimators': 400}
0.683 (+/-0.105) for {'max_depth': 20, 'n_estimators': 100}
0.645 (+/-0.148) for {'max_depth': 20, 'n_estimators': 200}
0.655 (+/-0.148) for {'max_depth': 20, 'n_estimators': 400}

count 105/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.69      0.67      0.68        99
           1       0.69      0.71      0.70       101

    accuracy                           0.69       200
   macro avg       0.69      0.69      0.69       200
weighted avg       0.69 

### Cleaned Data

In [16]:
X, y, feature_list = convert2np(data_df_nomiss, lbl_colname, selected_feat=True)

In [17]:
hfpef_rf(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'max_depth': 5, 'n_estimators': 100}

Grid scores on development set:

0.463 (+/-0.015) for {'max_depth': 5, 'n_estimators': 100}
0.459 (+/-0.018) for {'max_depth': 5, 'n_estimators': 200}
0.459 (+/-0.018) for {'max_depth': 5, 'n_estimators': 400}
0.463 (+/-0.015) for {'max_depth': 10, 'n_estimators': 100}
0.459 (+/-0.018) for {'max_depth': 10, 'n_estimators': 200}
0.459 (+/-0.018) for {'max_depth': 10, 'n_estimators': 400}
0.463 (+/-0.015) for {'max_depth': 20, 'n_estimators': 100}
0.459 (+/-0.018) for {'max_depth': 20, 'n_estimators': 200}
0.459 (+/-0.018) for {'max_depth': 20, 'n_estimators': 400}

count 0/80
Detailed classification report:

              precision    recall  f1-score   support

           0       0.86      1.00      0.93        69
           1       0.00      0.00      0.00        11

    accuracy                           0.86        80
   macro avg       0.43      0.50      0.46        80
weighte

C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\model_select

Best parameters set found on development set:

{'max_depth': 5, 'n_estimators': 100}

Grid scores on development set:

0.587 (+/-0.413) for {'max_depth': 5, 'n_estimators': 100}
0.587 (+/-0.413) for {'max_depth': 5, 'n_estimators': 200}
0.587 (+/-0.413) for {'max_depth': 5, 'n_estimators': 400}
0.587 (+/-0.413) for {'max_depth': 10, 'n_estimators': 100}
0.587 (+/-0.413) for {'max_depth': 10, 'n_estimators': 200}
0.587 (+/-0.413) for {'max_depth': 10, 'n_estimators': 400}
0.587 (+/-0.413) for {'max_depth': 20, 'n_estimators': 100}
0.587 (+/-0.413) for {'max_depth': 20, 'n_estimators': 200}
0.587 (+/-0.413) for {'max_depth': 20, 'n_estimators': 400}



C:\Users\HP\miniconda3\lib\site-packages\sklearn\model_selection\_split.py:666: UserWarning: The least populated class in y has only 4 members, which is less than n_splits=5.
  warnings.warn(("The least populated class in y has only %d"


count 0/80
Detailed classification report:

              precision    recall  f1-score   support

           0       0.95      1.00      0.97        76
           1       0.00      0.00      0.00         4

    accuracy                           0.95        80
   macro avg       0.47      0.50      0.49        80
weighted avg       0.90      0.95      0.93        80


Feature: NTProBNP, Score: 0.14208
Feature: Cr, Score: 0.13959
Feature: Age, Score: 0.11756
Feature: medial E', Score: 0.11704
Feature: LAVI, Score: 0.11370
Feature: MAP, Score: 0.08938
Feature: GFR, Score: 0.08658
Feature: LA diameter, Score: 0.05525
Feature: PP, Score: 0.05519
Feature: medial a', Score: 0.04839
Feature: CKD stage, Score: 0.02700
Feature: Sex, Score: 0.00476
Feature: AF, Score: 0.00349
Prediction of Major cardiac events


C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Best parameters set found on development set:

{'max_depth': 5, 'n_estimators': 100}

Grid scores on development set:

0.574 (+/-0.091) for {'max_depth': 5, 'n_estimators': 100}
0.537 (+/-0.074) for {'max_depth': 5, 'n_estimators': 200}
0.501 (+/-0.167) for {'max_depth': 5, 'n_estimators': 400}
0.541 (+/-0.143) for {'max_depth': 10, 'n_estimators': 100}
0.545 (+/-0.150) for {'max_depth': 10, 'n_estimators': 200}
0.526 (+/-0.215) for {'max_depth': 10, 'n_estimators': 400}
0.541 (+/-0.143) for {'max_depth': 20, 'n_estimators': 100}
0.545 (+/-0.150) for {'max_depth': 20, 'n_estimators': 200}
0.526 (+/-0.215) for {'max_depth': 20, 'n_estimators': 400}

count 64/80
Detailed classification report:

              precision    recall  f1-score   support

           0       0.56      0.31      0.40        29
           1       0.69      0.86      0.77        51

    accuracy                           0.66        80
   macro avg       0.62      0.59      0.58        80
weighted avg       0.64   

##  Neural Networks

In [18]:
from sklearn.neural_network import MLPClassifier
from sklearn import preprocessing

In [19]:
def hfpef_ann(X, y, feature_list):
    # Set the parameters by cross-validation
    tuned_parameters = [{'hidden_layer_sizes': [(20,20,20), (100),(20,20)]}]

    ann = MLPClassifier(activation='tanh', max_iter=1000, random_state=0)

    for i in range(y.shape[1]):
        print('==================================================')
        print('Prediction of ' + lbl_colname[i])

        y_true = y[:,i]

        clf = GridSearchCV(ann, tuned_parameters, scoring='f1_macro', cv=5)
        clf.fit(X, y_true)

        print("Best parameters set found on development set:")
        print()
        print(clf.best_params_)
        print()
        print("Grid scores on development set:")
        print()
        means = clf.cv_results_['mean_test_score']
        stds = clf.cv_results_['std_test_score']
        for mean, std, params in zip(means, stds, clf.cv_results_['params']):
            print("%0.3f (+/-%0.03f) for %r"
                  % (mean, std * 2, params))
        print()

        y_pred = cross_val_predict(clf.best_estimator_, X, y_true, cv=5)

        print(f"count {sum(y_pred)}/{len(y_pred)}")
        print("Detailed classification report:")
        print()
        print(classification_report(y_true, y_pred))
        print()

### All Features

In [42]:
X, y, feature_list = convert2np(data_df, lbl_colname)

from sklearn.decomposition import PCA

X = PCA(n_components=0.95).fit_transform(X)
# scaler = preprocessing.MinMaxScaler().fit(X)
# X = scaler.transform(X)

In [43]:
X.shape

(200, 1)

In [44]:
hfpef_ann(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'hidden_layer_sizes': 100}

Grid scores on development set:

0.458 (+/-0.006) for {'hidden_layer_sizes': (20, 20, 20)}
0.505 (+/-0.193) for {'hidden_layer_sizes': 100}
0.458 (+/-0.006) for {'hidden_layer_sizes': (20, 20)}

count 44/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.85      0.79      0.82       169
           1       0.18      0.26      0.21        31

    accuracy                           0.70       200
   macro avg       0.52      0.52      0.52       200
weighted avg       0.75      0.70      0.72       200


Prediction of CV death
Best parameters set found on development set:

{'hidden_layer_sizes': 100}

Grid scores on development set:

0.486 (+/-0.005) for {'hidden_layer_sizes': (20, 20, 20)}
0.489 (+/-0.154) for {'hidden_layer_sizes': 100}
0.486 (+/-0.005) for {'hidden_layer_sizes': (20, 20)}

count 44/200
Detailed classification repo

==================================================
Prediction of Death
Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.519 (+/-0.125) for {'hidden_layer_sizes': (20, 20, 20)}
0.473 (+/-0.071) for {'hidden_layer_sizes': 100}
0.458 (+/-0.006) for {'hidden_layer_sizes': (20, 20)}

count 11/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.85      0.95      0.90       169
           1       0.27      0.10      0.14        31

    accuracy                           0.82       200
   macro avg       0.56      0.52      0.52       200
weighted avg       0.76      0.82      0.78       200


==================================================
Prediction of CV death
Best parameters set found on development set:

{'hidden_layer_sizes': 100}

Grid scores on development set:

0.485 (+/-0.007) for {'hidden_layer_sizes': (20, 20, 20)}
0.486 (+/-0.005) for {'hidden_layer_sizes': 100}
0.486 (+/-0.005) for {'hidden_layer_sizes': (20, 20)}

count 0/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.94      1.00      0.97       189
           1       0.00      0.00      0.00        11

    accuracy                           0.94       200
   macro avg       0.47      0.50      0.49       200
weighted avg       0.89      0.94      0.92       200


==================================================
Prediction of Major cardiac events

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.685 (+/-0.132) for {'hidden_layer_sizes': (20, 20, 20)}
0.661 (+/-0.076) for {'hidden_layer_sizes': 100}
0.666 (+/-0.103) for {'hidden_layer_sizes': (20, 20)}

count 111/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.71      0.64      0.67        99
           1       0.68      0.74      0.71       101

    accuracy                           0.69       200
   macro avg       0.69      0.69      0.69       200
weighted avg       0.69      0.69      0.69       200

### Selected Features

In [45]:
X, y, feature_list = convert2np(data_df, lbl_colname, selected_feat=True)

X = PCA(n_components=0.95).fit_transform(X)

# scaler = preprocessing.MinMaxScaler().fit(X)
# X = scaler.transform(X)

In [46]:
hfpef_ann(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'hidden_layer_sizes': 100}

Grid scores on development set:

0.458 (+/-0.006) for {'hidden_layer_sizes': (20, 20, 20)}
0.505 (+/-0.193) for {'hidden_layer_sizes': 100}
0.458 (+/-0.006) for {'hidden_layer_sizes': (20, 20)}

count 44/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.85      0.79      0.82       169
           1       0.18      0.26      0.21        31

    accuracy                           0.70       200
   macro avg       0.52      0.52      0.52       200
weighted avg       0.75      0.70      0.72       200


Prediction of CV death
Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.486 (+/-0.005) for {'hidden_layer_sizes': (20, 20, 20)}
0.485 (+/-0.145) for {'hidden_layer_sizes': 100}
0.486 (+/-0.005) for {'hidden_layer_sizes': (20, 20)}

count 0/200
Detailed classificat

C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.603 (+/-0.135) for {'hidden_layer_sizes': (20, 20, 20)}
0.603 (+/-0.135) for {'hidden_layer_sizes': 100}
0.603 (+/-0.135) for {'hidden_layer_sizes': (20, 20)}

count 44/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.58      0.92      0.71        99
           1       0.82      0.36      0.50       101

    accuracy                           0.64       200
   macro avg       0.70      0.64      0.61       200
weighted avg       0.70      0.64      0.60       200




### Cleaned Data

In [24]:
X, y, feature_list = convert2np(data_df_nomiss, lbl_colname, selected_feat=True)

# scaler = preprocessing.StandardScaler().fit(X)
# X = scaler.transform(X)

In [25]:
hfpef_ann(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20)}

Grid scores on development set:

0.459 (+/-0.018) for {'hidden_layer_sizes': (20, 20, 20)}
0.459 (+/-0.018) for {'hidden_layer_sizes': 100}
0.463 (+/-0.015) for {'hidden_layer_sizes': (20, 20)}

count 0/80
Detailed classification report:

              precision    recall  f1-score   support

           0       0.86      1.00      0.93        69
           1       0.00      0.00      0.00        11

    accuracy                           0.86        80
   macro avg       0.43      0.50      0.46        80
weighted avg       0.74      0.86      0.80        80


Prediction of CV death


C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\HP\miniconda3\lib\site-packages\sklearn\model_select

Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.584 (+/-0.417) for {'hidden_layer_sizes': (20, 20, 20)}
0.584 (+/-0.417) for {'hidden_layer_sizes': 100}
0.584 (+/-0.417) for {'hidden_layer_sizes': (20, 20)}



C:\Users\HP\miniconda3\lib\site-packages\sklearn\model_selection\_split.py:666: UserWarning: The least populated class in y has only 4 members, which is less than n_splits=5.
  warnings.warn(("The least populated class in y has only %d"


count 1/80
Detailed classification report:

              precision    recall  f1-score   support

           0       0.95      0.99      0.97        76
           1       0.00      0.00      0.00         4

    accuracy                           0.94        80
   macro avg       0.47      0.49      0.48        80
weighted avg       0.90      0.94      0.92        80


Prediction of Major cardiac events
Best parameters set found on development set:

{'hidden_layer_sizes': 100}

Grid scores on development set:

0.384 (+/-0.030) for {'hidden_layer_sizes': (20, 20, 20)}
0.486 (+/-0.191) for {'hidden_layer_sizes': 100}
0.421 (+/-0.123) for {'hidden_layer_sizes': (20, 20)}

count 65/80
Detailed classification report:

              precision    recall  f1-score   support

           0       0.40      0.21      0.27        29
           1       0.65      0.82      0.72        51

    accuracy                           0.60        80
   macro avg       0.52      0.52      0.50        80
weigh